I would also like to add some instructions for getting started. That may currently be blocked by a few things, but perhaps we could start with what dependencies need to be installed. In trying to do the normal build process with `./configure` then `make world.opt`, I am getting this:
```sh
...
gcc -c -O2 -fno-strict-aliasing -fwrapv -Wall -Werror -fno-tree-vrp  -g -DDEBUG -D_FILE_OFFSET_BITS=64 -D_REENTRANT -DCAML_NAME_SPACE  -o instrtrace.d.o instrtrace.c
rm -f libcamlruni.a && emar rc libcamlruni.a  interp.i.o misc.i.o stacks.i.o fix_code.i.o startup_aux.i.o startup.i.o freelist.i.o major_gc.i.o minor_gc.i.o memory.i.o alloc.i.o roots.i.o globroots.i.o fail.i.o signals.i.o signals_byt.i.o printexc.i.o backtrace_prim.i.o backtrace.i.o compare.i.o ints.i.o floats.i.o str.i.o array.i.o io.i.o extern.i.o intern.i.o hash.i.o sys.i.o meta.i.o parsing.i.o gc_ctrl.i.o md5.i.o obj.i.o lexing.i.o callback.i.o debugger.i.o weak.i.o compact.i.o finalise.i.o custom.i.o dynlink.i.o spacetime.i.o afl.i.o unix.i.o bigarray.i.o main.i.o; emranlib libcamlruni.a libcamlruni.a
(echo '#define CAML_INTERNALS'; \
         echo '#include "caml/mlvalues.h"'; \
 echo '#include "caml/prims.h"'; \
 sed -e 's/.*/extern value &();/' primitives; \
 echo 'c_primitive caml_builtin_cprim[] = {'; \
 sed -e 's/.*/	&,/' primitives; \
 echo '	 0 };'; \
 echo 'char * caml_names_of_builtin_cprim[] = {'; \
 sed -e 's/.*/	"&",/' primitives; \
 echo '	 0 };') > prims.c
/bin/sh: 1: emar: not found
/bin/sh: 1: emranlib: not found
Makefile:200: recipe for target 'libcamlruni.a' failed
make[1]: *** [libcamlruni.a] Error 127
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/home/zach/src/ocaml-wasm/byterun'
Makefile:413: recipe for target 'coldstart' failed
make: *** [coldstart] Error 2
```
So we at least need `emscripten` available (not sure on version requirements), and apparently a fork of LLVM.

--- a/asmcomp/wasm32/README.md
+++ b/asmcomp/wasm32/README.md
@@ -8,17 +8,21 @@ cmm -> typed cmm -> wasm ast -> linking -> binary
 Typed CMM
 ---
 
-In normal OCaml compilation, after typechecking has been completed the types are removed,
-and thus the cmm representation does not include types.
-However, for WASM, adding some types back at this stage is helpful to cleanly generate the WebAssembly,
-because CMM is closer to WASM than it is to the typed AST.
-WebAssembly types and helper functions are defined in [typed_cmm.ml](typed_cmm.ml).
+In normal OCaml compilation, after typechecking has been completed most types are removed.
+However, for WASM, including some additional type information at this stage is helpful
+to generate the types required by WebAssembly.
+Type enhancements to the CMM provided by OCaml are defined in [typed_cmm.ml](typed_cmm.ml).
+`Typed_cmm` is used in [emit.mlp](emit.mlp) to translate that CMM to the WebAssembly AST,
+which then gets transformed in actual binary code in [encode.ml](encode.ml).
 
 Linking
 ---
 
-We currently depend on a fork of LLVM for linking with other WASM code.
+We currently depend on LLVM for linking with other WASM code.
 This is desirable for interoperability with code written in other languages.
+
+*NOTE:* The `wasm-backend` branch currently depends on a fork of LLVM to support garbage collection.
+The intention is to eventually upstream these changes into LLVM.
 
 Contributing
 ---
@@ -33,7 +37,7 @@ Modules
 ---
 
 The entry-point for WASM compilation is [asmcomp/asmgen.mlp-wasm](../asmgen.mlp-wasm).
-The main points of interest in the pipline are:
+The main points of interest in the pipeline are:
 
 [Typed_cmm](Typed_cmm.ml): Adds types to the CMM representation.
 

